### PR TITLE
React to hosting

### PR DESCRIPTION
**Pull request by HaoK**
https://github.com/aspnet/Hosting/pull/210

cc @divega

### CODE REVIEW

**Comment by divega** [comment]
:shipit:

**Comment by bricelam** [comment]
Cool; much easier. :shipit:

**Comment by HaoK** [comment]
Updated to use new overload, now even cleaner, one line with no arguments :)

File: src/EntityFramework.Commands/ContextTool.cs [hunk 1]
```diff
@@ -37,12 +34,7 @@ private static DbContext TryCreateContextFromStartup(Type type)
 #if DNX451 || DNXCORE50
             try
             {
-                var context = new HostingContext
-                {
-                    ServerFactory = new ServerFactory(),
-                };
-                var instance = new HostingEngine().Start(context);
-                return context.ApplicationServices.GetService(type) as DbContext;
+                return WebApplication.CreateHostingEngine().ApplicationServices.GetService(type) as DbContext;
```
Comment (davidfowl) [inline src/EntityFramework.Commands/ContextTool.cs]: Holy one liner :smile:

Comment (divega) [inline src/EntityFramework.Commands/ContextTool.cs]: Not trying to be a wet blanket, but can you please split it? :smile:

Comment (divega) [inline src/EntityFramework.Commands/ContextTool.cs]: Not trying to be a wet blanket, but can you please split it? :smile:

**Comment by divega** [comment]
:shipit:

**Pull request closed without merging**